low-level/framework/utils/sysfs_interface.py

```diff
@@ -138,7 +140,144 @@ def fetch_nw_cable_status(self, nw_interface_path, interface):
                 carrier_indicator = cFile.read().strip()
             if carrier_indicator not in self.nw_phy_link_state.keys():
                 carrier_indicator = 'unknown'
-        except OSError as os_err:
-            return os_err.errno
+        except Exception:
+            carrier_indicator = 'unknown'
         return self.nw_phy_link_state[carrier_indicator]
 
+    def fetch_nw_operstate(self, interface):
+        """
+        Return the Operational Status of the Network Interface.
+
+        Possible Values: <'UNKNOWN', 'NOTPRESENT", 'DOWN', 'LOWERLAYERDOWN',
+                          'TESTING', 'DORMANT', 'UP'>
+        """
+        operstate_file_path = os.path.join(self.get_sys_dir_path('net'),
+                                           interface, 'operstate')
+        operstate = "UNKNOWN"
+        try:
+            if os.path.isfile(operstate_file_path):
+                with open(operstate_file_path) as cFile:
+                    operstate = cFile.read().strip().upper()
+        except Exception as err:
```

Comment:
wow, catch the exception and raise same again ? why to catch it then ... avoid it, caller will catch it at its end

Reply:
sure. I need not to catch any exception here in sysfs_interface then, since we don't want to error out of here instead just return the empty result.

Reply:
trying to understand expected behaviour, in case of Expection Handling, added comment in new PR.

Reply:
Set `operstate = "UP"` before try block.  `operstate = "UNKNOWN"` inside except block.
In case os/file operation fails, unknown could be reasonable.